contracts/testContracts/TestFlowCallbackHandler.cdc

```diff
@@ -0,0 +1,107 @@
+            self.name = name
+            self.description = description
+        }
+
+        access(all) fun getName(): String {
+            return self.name
+        }
+
+        access(all) fun getDescription(): String {
+            return self.description
+        }
+
+        access(all) view fun getViews(): [Type] {
+            return []
+        }
+
+        access(all) fun resolveView(_ view: Type): AnyStruct? {
+            return nil
+        }
+        
+        access(FlowCallbackScheduler.Execute) 
+        fun executeCallback(id: UInt64, data: AnyStruct?) {
+            // Most callbacks will have string data
+            if let dataString = data as? String {
+                // intentional failure test case
+                if dataString == "fail" {
+                    panic("Callback \(id) failed")
+                } else if dataString == "cancel" {
+                    // This should always fail because the callback can't cancel itself during execution
+                    destroy <-TestFlowCallbackHandler.cancelCallback(id: id)
+                } else {
+                    // All other regular test cases should succeed
+                    TestFlowCallbackHandler.succeededCallbacks.append(id)
+                }
+            } else if let dataCap = data as? Capability<auth(FlowCallbackScheduler.Execute) &{FlowCallbackScheduler.CallbackHandler}> {
+                // Testing scheduling a callback with a callback
+                let scheduledCallback <- FlowCallbackScheduler.schedule(
+                    callback: dataCap,
+                    data: "test data",
+                    timestamp: getCurrentBlock().timestamp + 10.0,
+                    priority: FlowCallbackScheduler.Priority.High,
+                    executionEffort: UInt64(1000),
+                    fees: <-TestFlowCallbackHandler.getFeeFromVault(amount: 1.0)
+                )
+                TestFlowCallbackHandler.addScheduledCallback(callback: <-scheduledCallback)
+            } else {
+                panic("TestFlowCallbackHandler.executeCallback: Invalid data type for callback with id \(id). Type is \(data.getType().identifier)")
+            }
+        }
+    }
+
+    access(all) fun createHandler(): @Handler {
+        return <- create Handler(name: "Test FlowCallbackHandler Resource", description: "Executes a variety of callbacks for different test cases")
+    }
+
+    access(all) fun addScheduledCallback(callback: @FlowCallbackScheduler.ScheduledCallback) {
+        let status = callback.status()
+        if status == nil {
+            panic("Invalid status for callback with id \(callback.id)")
+        }
+        self.scheduledCallbacks[callback.id] <-! callback
+    }
+
+    access(all) fun cancelCallback(id: UInt64): @FlowToken.Vault {
+        let callback <- self.scheduledCallbacks.remove(key: id)
+            ?? panic("Invalid ID: \(id) callback not found")
+        return <-FlowCallbackScheduler.cancel(callback: <-callback!)
+    }
```

Comment:
I think comments related to access would be helpful here. In general, a contract dev wouldn't want to expose these methods as public